components/login.jsx

```diff
@@ -0,0 +1,112 @@
+var React = require('react');
+var Router = require('react-router');
+var Link = Router.Link;
+
+var teachAPI = require('../lib/teach-api');
+
+var Login = React.createClass({
+  getDefaultProps: function() {
+    return {
+      teachAPI: teachAPI,
+      alert: defaultAlert
```

Comment:
I tend to put function declarations below variable declarations, so it's defined near the end of this file, but we can either move the function declaration up, or define the function as a variable above the component if you want.